frontend/src/App.js

```diff
@@ -1,9 +1,40 @@
 import React from 'react'
+import { BrowserRouter, Route, Switch } from 'react-router-dom'
+import { Provider } from 'react-redux'
+import { configureStore, combineReducers } from '@reduxjs/toolkit'
+
+import Header from './components/Header'
+import Home from './Pages/Home'
+import SignUp from './Pages/SignUp'
+import Profile from './Pages/Profile'
+
+import user from './reducers/user'
+
+const reducer = combineReducers({
+  user: user.reducer
+})
+
+const store = configureStore({ reducer })
 
 export const App = () => {
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
+    <BrowserRouter>
+      <Provider store={store}>  
+        <Header />
+        <main>
+          <Switch>
+            <Route exact path='/'>
```

Comment:
If the user types in their credentials for an unexisting account and press signup, one could expect to automatically send a signup request instead of navigating to a signup page, where credentials need to be written again.